### PR TITLE
`/get_ursulas` should perform node sampling using `TACoChildApplication` instead of `TACoApplication`

### PR DESCRIPTION
**Pull request by derekpierre**
**Before merging** update the `nucypher` dependency to not point to the branch used for PR #3341 development.
=========

**Type of PR:**
- [ ] Bugfix
- [X] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [ ] 2
- [X] 3

**What this does:**
Related to:
- https://github.com/nucypher/nucypher-contracts/pull/171

Depends on:
- https://github.com/nucypher/nucypher/pull/3341

Until we have better syncing between root and child applications, `TACoChildApplication` has the most up-to-date information for stakers , and so should be used for sampling nodes instead.

**NOTE:** once these child sampling PRs merged, TACo contracts need to be redeployed on testnet to the latest in `nucypher-contracts`; ideally only `lynx` to start, since redeployment implies new rituals need to be created because Coordinator was not previously updateable.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR addresses a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

**Review by piotr-roslaniec — APPROVED** [review]

File: Pipfile [hunk 1]
```diff
@@ -7,13 +7,13 @@ name = "pypi"
 python_version = "3"
 
 [packages]
-nucypher = {git = "https://github.com/nucypher/nucypher.git", ref = "v7.0.0-rc.8"}
+nucypher = {git = "https://github.com/derekpierre/nucypher.git", ref = "child-sampling"}
```
Comment (cygnusv) [inline Pipfile:10]:
```suggestion
nucypher = {git = "https://github.com/nucypher/nucypher.git", ref = "v7.0.0-rc.10"}
```

File: Pipfile [hunk 2]
```diff
@@ -7,13 +7,13 @@ name = "pypi"
...
 nucypher-core = "==0.13.0"  # must be the same as nucypher
 flask-cors = "*"
 prometheus-flask-exporter = "*"
 
 [dev-packages]
-nucypher = {git = "https://github.com/nucypher/nucypher.git", editable = true, ref = "v7.0.0-rc.8", extras = ["dev"]}  # needed for testerchain, and must be editable
+nucypher = {git = "https://github.com/derekpierre/nucypher.git", editable = true, ref = "child-sampling", extras = ["dev"]}  # needed for testerchain, and must be editable
```
Comment (cygnusv) [inline Pipfile:16]:
```suggestion
nucypher = {git = "https://github.com/nucypher/nucypher.git", editable = true, ref = "v7.0.0-rc.10", extras = ["dev"]}
```

File: dev-requirements.txt [hunk 3]
```diff
@@ -83,7 +83,7 @@ msgspec==0.18.4 ; python_version >= '3.8'
 multidict==5.2.0 ; python_version >= '3.6'
 mypy-extensions==1.0.0 ; python_version >= '3.5'
 nodeenv==1.8.0 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6'
-git+https://github.com/nucypher/nucypher.git@598ad313762fa2a5c8653f5996e888e75d9370ed#egg=nucypher
+git+https://github.com/derekpierre/nucypher.git@f751cb385e03909529c733a402b674c7c2f28cdd#egg=nucypher
```
Comment (cygnusv) [inline dev-requirements.txt:86]: Need to relock dependencies after new release of nucypher

File: porter/cli/main.py [hunk 4]
```diff
@@ -30,8 +30,15 @@ def porter_cli():
...
 @porter_cli.command()
 @group_general_config
-@option_domain(default=str(domains.DEFAULT_DOMAIN), validate=True, required=False)
-@option_eth_endpoint(required=False)
+@option_domain(default=str(domains.DEFAULT_DOMAIN), validate=True, required=True)
+@option_eth_endpoint(required=True)
+@click.option(
+    "--polygon-endpoint",
+    "polygon_endpoint",
+    help="Connection URL for PRE payment method",
```
Comment (cygnusv) [inline porter/cli/main.py:38]:
```suggestion
    help="Connection URL for Polygon chain",
```

File: porter/main.py [hunk 5]
```diff
@@ -125,6 +123,22 @@ def __init__(
 
         self.log.info(BANNER)
 
+    @staticmethod
+    def _initialize_endpoints(eth_endpoint: str, polygon_endpoint: str):
+        if not eth_endpoint:
```
Comment (cygnusv) [inline porter/main.py:128]: I guess this is not reachable if Porter is launched via the CLI, but for cases when it's instantiated programatically?

File: requirements.txt [hunk 6]
```diff
@@ -53,7 +53,7 @@ msgpack==1.0.7 ; python_version >= '3.8'
 msgpack-python==0.5.6
 multidict==5.2.0 ; python_version >= '3.6'
 mypy-extensions==1.0.0 ; python_version >= '3.5'
-git+https://github.com/nucypher/nucypher.git@598ad313762fa2a5c8653f5996e888e75d9370ed#egg=nucypher
+git+https://github.com/derekpierre/nucypher.git@f751cb385e03909529c733a402b674c7c2f28cdd#egg=nucypher
```
Comment (cygnusv) [inline requirements.txt:56]: Relock

File: tests/pre/conftest.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    ):
+        addresses = {
+            address: 1
+            for address in testerchain.stake_providers_accounts
+            if address not in without
+        }
+        return StakingProvidersReservoir(addresses)
+
+    # TODO - this is needed for PRE Policy.enact(...) sample functionality which
+    #  uses TACoApplication - should we change this (in `nucypher`)?
```
Comment (cygnusv) [inline tests/pre/conftest.py:24]: Yes, but it has low priority for the moment. Opened https://github.com/nucypher/nucypher/issues/3346

**Review by cygnusv — APPROVED** [review]
LGTM, but needs dependency updates after nucypher gets a new release

File: tests/pre/conftest.py (hunk 7 above)
Comment (derekpierre) [reply tests/pre/conftest.py:24]: Thanks for filing 🙏

File: porter/main.py (hunk 5 above)
Comment (derekpierre) [reply porter/main.py]: Yes. However, perform the None check in the __init__ now along with some other clean up of parameters.

**Pull request merged**